guppylang/prelude/quantum.py

```diff
@@ -26,84 +26,209 @@ def __new__() -> "qubit":
         reset(q)
         return q
 
+    @guppy
+    @no_type_check
+    def h(self: "qubit") -> None:
+        h(self)
+
+    @guppy
+    @no_type_check
+    def cz(self: "qubit", target: "qubit") -> None:
+        cz(self, target)
+
+    @guppy
+    @no_type_check
+    def cx(self: "qubit", target: "qubit") -> None:
```

Comment:
I would suggest`target.cx(control)` instead,
so the n-controlled cases keep the same structure.
```py
target.toffoli(control0, control1)
target.ncx(*controls)
```

Reply:
Whichever way round it is, I see a lot of potential for bugs caused by people getting it the wrong way round.
In general I'm not so keen on forcing multi-qubit operations to have a "distinguished" qubit in this way: in general there's no natural way to select which qubit it should be.

Reply:
I think that's very valid. How about we restrict it a specific set of special single qubit operations: alloc (already the case), discard, measure (non destructive), reset, and any variants of those above (e.g. measure_reset, destructive measure). These special set can be implemented in terms of the existing inout functions as in this PR currently.

Reply:
Done